com.civicrm.hrjobroles/js/src/job-roles/services/hr-job-roles-service.js

```diff
@@ -17,12 +16,64 @@ define([
                     "contact_id": contact_id,
                     "deleted": 0,
                     "return": "title,period_end_date,period_start_date"
-                }).done(function(result) {
+                }).done(function (result) {
+                    //get revisions
+                    var revisions = [];
+                    result.values.forEach(function (contract) {
+                        revisions.push(CRM.api3('HRJobContractRevision', 'get', {
+                            "sequential": 1,
+                            "jobcontract_id": contract.id
+                        }).then(function (response) {
+                            return response.values.map(function (item) {
+                                return {
+                                    id: item.id,
+                                    contract_id: item.jobcontract_id
+                                }
+                            });
+                        }));
+                    });
 
-                    // Passing data to deferred's resolve function on successful completion
-                    deferred.resolve(result);
 
-                }).error(function(result) {
+                    $q.all(revisions).then(function (response) {
```

Comment:
This here is a very long chain, it would be best explaining what each step (`.then()`) does with a comment